bin/inx-chronicle/src/stardust_inx/mod.rs

```diff
@@ -122,6 +123,17 @@ impl Actor for InxWorker {
         let mut inx_client = Self::connect(&self.config).await?;
         log::info!("Connected to INX.");
 
+        log::info!("Reading unspent outputs.");
+        let mut unspent_output_stream = inx_client.read_unspent_outputs(NoParams {}).await?.into_inner();
+
+        let mut updates = Vec::new();
+        while let Some(unspent_output) = unspent_output_stream.try_next().await? {
+            let ledger_output: inx::LedgerOutput = unspent_output.output.unwrap().try_into()?;
+            updates.push(ledger_output.into());
+        }
+        log::info!("Inserting {} unspent outputs.", updates.len());
+        self.db.insert_ledger_updates(updates).await?;
```

Comment:
I think I remember from the call that first buffering all ledger updates in a Vec happens on purpose, but I am not sure and I am not sure whether it's obvious. Could you add a comment why we're not inserting them immediatedly?

Reply:
Because they come from a stream, not an iterator.